src/libraries/System.Security.Cryptography.Encoding/tests/Cbor.Tests/CoseKeyHelpers.cs

```diff
@@ -0,0 +1,261 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.Security.Cryptography;
+
+namespace System.Formats.Cbor.Tests
+{
+    public static class CborCoseKeyHelpers
+    {
+        public static byte[] ExportCosePublicKey(ECDsa ecDsa, HashAlgorithmName hashAlgName)
+        {
+            ECParameters ecParams = ecDsa.ExportParameters(includePrivateParameters: false);
+            using var writer = new CborWriter(CborConformanceLevel.Ctap2Canonical);
+            WriteECParametersAsCosePublicKey(writer, ecParams, hashAlgName);
+            return writer.GetEncoding();
+        }
+
+        public static (ECDsa ecDsa, HashAlgorithmName hashAlgName) ParseCosePublicKey(byte[] coseKey)
+        {
+            var reader = new CborReader(coseKey, CborConformanceLevel.Ctap2Canonical);
+            (ECParameters ecParams, HashAlgorithmName hashAlgName) = ReadECParametersAsCosePublicKey(reader);
+            return (ECDsa.Create(ecParams), hashAlgName);
+        }
+
+        public static void WriteECParametersAsCosePublicKey(CborWriter writer, ECParameters ecParams, HashAlgorithmName algorithmName)
+        {
+            Debug.Assert(writer.ConformanceLevel == CborConformanceLevel.Ctap2Canonical);
+
+            if (ecParams.Q.X is null || ecParams.Q.Y is null)
+            {
+                throw new ArgumentException("does not specify a public key point.", nameof(ecParams));
+            }
+
+            // run these first to perform necessary validation
+            (CoseKeyType kty, CoseCrvId crv) = MapECCurveToCoseKtyAndCrv(ecParams.Curve);
+            CoseKeyAlgorithm alg = MapHashAlgorithmNameToCoseKeyAlg(algorithmName);
+
+            // begin writing CBOR object
+            writer.WriteStartMap();
+
+            WriteCoseKeyLabel(writer, CoseKeyLabel.Kty);
+            writer.WriteInt32((int)kty);
+
+            WriteCoseKeyLabel(writer, CoseKeyLabel.Alg);
+            writer.WriteInt32((int)alg);
+
+            WriteCoseKeyLabel(writer, CoseKeyLabel.EcCrv);
+            writer.WriteInt32((int)crv);
+
+            WriteCoseKeyLabel(writer, CoseKeyLabel.EcX);
+            writer.WriteByteString(ecParams.Q.X);
+
+            WriteCoseKeyLabel(writer, CoseKeyLabel.EcY);
+            writer.WriteByteString(ecParams.Q.Y);
+
+            writer.WriteEndMap();
+
+            static (CoseKeyType kty, CoseCrvId crv) MapECCurveToCoseKtyAndCrv(ECCurve curve)
```

Comment:
I think it would be easier to read this without abbreviating names (destructed variable names and function name)